geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoAttributeKeyValueIterator.scala

```diff
@@ -46,14 +46,16 @@ class KryoAttributeKeyValueIterator extends SortedKeyValueIterator[Key, Value] w
     IteratorClassLoader.initClassLoader(getClass)
 
     this.source = src.deepCopy(env)
-    sft = SimpleFeatureTypes.createType("", options.get(SFT_OPT))
+
+    val spec = options.get(SFT_OPT)
+    sft = IteratorCache.sft(spec)
     attribute = options.get(ATTRIBUTE_OPT).toInt
 
     val index = try { AccumuloFeatureIndex.index(options.get(INDEX_OPT)) } catch {
       case NonFatal(e) => throw new RuntimeException(s"Index option not configured correctly: ${options.get(INDEX_OPT)}")
     }
     val kryoOptions = if (index.serializedWithId) SerializationOptions.none else SerializationOptions.withoutId
-    serializer = new KryoFeatureSerializer(sft, kryoOptions)
+    serializer = IteratorCache.serializer(spec, kryoOptions)
```

Comment:
hmm, the options are different for different iterators so that might be more trouble than it's worth.